client/lib/sites-list/list.js

```diff
@@ -400,12 +439,18 @@ SitesList.prototype.setRecentlySelectedSite = function( siteID ) {
 		return;
 	}
 
+	const index = this.recentlySelected.indexOf( siteID );
+
 	// do not add duplicates
-	if ( this.recentlySelected.indexOf( siteID ) === -1 ) {
-		this.recentlySelected.unshift( siteID );
+	if ( index === -1 ) {
+		if ( ! this.isStarred( this.getSite( siteID ) ) ) {
+			this.recentlySelected.unshift( siteID );
+		}
 	} else {
-		this.recentlySelected.splice( this.recentlySelected.indexOf( siteID ), 1 );
-		this.recentlySelected.unshift( siteID );
+		this.recentlySelected.splice( index, 1 );
+		if ( ! this.isStarred( this.getSite( siteID ) ) ) {
```

Comment:
Why do we need to check `isStarred` here? I don't think recently selected should care whether the site is starred, and you're already already [filtering out starred sites](https://github.com/Automattic/wp-calypso/blob/ba66b4eae18df0f4a19a4e40df034c1e8b09c4b5/client/components/site-selector/index.jsx#L217) in the `renderRecentSites` anyways.